rollup-operator/src/synch.js

```diff
@@ -134,29 +143,30 @@ class Synchronizer {
     }
 
     async getLastBatch(){
-        return Number(this._fromString(await this.db.getOrDefault(lastBatchKey, "0")));
+        return Number(this._fromString(await this.db.getOrDefault(lastBatchKey, "-1")));
     }
 
-    async _updateEvents(logs, lastBatchSaved, currentBatchDepth, blockNumber){
+    async _updateEvents(logs, nextBatchSynched, blockNumber){
         // save events on database
         logs.forEach((elem, index )=> {
             this._saveEvents(elem, index);
         });
         // Update rollupTree and last batch synched
-        for (let i = lastBatchSaved + 1; i <= currentBatchDepth; i++) {
-            const eventForge = [];
-            const eventOnChain = [];
-            // add off-chain tx
-            const keysForge = await this.db.listKeys(`${eventForgeBatchKey}${separator}${i-1}`);
-            for (const key of keysForge) eventForge.push(this._fromString(await this.db.get(key)));
-            // add on-chain tx
-            const keysOnChain = await this.db.listKeys(`${eventOnChainKey}${separator}${i-2}`);
-            for (const key of keysOnChain) eventOnChain.push(this._fromString(this.db.get(key)));
-            // Add events to rollup-tree
+        const batchKey = nextBatchSynched + 1;
+        const eventForge = [];
+        const eventOnChain = [];
+        // add off-chain tx
+        const keysForge = await this.db.listKeys(`${eventForgeBatchKey}${separator}${batchKey-1}`);
+        for (const key of keysForge) eventForge.push(this._fromString(await this.db.get(key)));
```

Comment:
Suggestion: const eventForge = keysForge.map(key => this._fromString(await this.db.get(key)));

Reply:
`map()` requires to solve all promises with `Promise.all()` 
It adds complex code readability therefore I will stay with the current version
:smile_cat: